core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -253,7 +253,7 @@ private DocumentMapper merge(DocumentMapper mapper, boolean updateAllTypes) {
                 // simulate first
                 MergeResult result = oldMapper.merge(mapper.mapping(), true, updateAllTypes);
                 if (result.hasConflicts()) {
-                    throw new MergeMappingException(result.buildConflicts());
+                    throw new IllegalArgumentException("Merge failed with failures {" + Arrays.toString(result.buildConflicts()) + "}");
```

Comment:
I just tried it and LoggerMessageFormat interprets the array as a var-arg, so only the first parameter is used: `System.out.println(LoggerMessageFormat.format("a {} d", new String[] {"b", "c"}));` prints "a b d".